src/scripts/Update.ts

```diff
@@ -1897,6 +1897,15 @@ class Update implements Saveable {
         },
 
         '0.10.6': ({ playerData, saveData }) => {
+            // Give the player any missing questline or temporary battle rewards
+            Update.giveMissingQuestLineProgressRewardPokemon(saveData, 'Unfinished Business', 8, 172.01);
+            Update.giveMissingQuestLineProgressRewardPokemon(saveData, 'Princess Diancie', 6, 681.01);
+            Update.giveMissingQuestLineProgressRewardPokemon(saveData, 'A Mystery Gift', 1, 801.01);
+            Update.giveMissingTempBattleRewardPokemon(saveData, 'Ash Ketchum Pinkan', 25.14);
+            Update.giveMissingTempBattleRewardPokemon(saveData, 'Ash Ketchum Alola', 25.08);
+            if (saveData.statistics.dungeonsCleared[GameConstants.getDungeonIndex('Tower of Waters')] > 0) {
```

Comment:
Unfortunately we need to use magic numbers in here, as the index could change in the future.